src/main/java/com/example/onlinebookshop/services/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.example.onlinebookshop.services.impl;
+
+import static com.example.onlinebookshop.constants.dtoconstants.CartItemDtoConstants.QUANTITY_DESCRIPTION;
+
+import com.example.onlinebookshop.dto.cartitem.request.AddCartItemDto;
+import com.example.onlinebookshop.dto.cartitem.request.UpdateItemQuantityDto;
+import com.example.onlinebookshop.dto.shoppingcart.response.ShoppingCartDto;
+import com.example.onlinebookshop.entities.Book;
+import com.example.onlinebookshop.entities.CartItem;
+import com.example.onlinebookshop.entities.ShoppingCart;
+import com.example.onlinebookshop.entities.User;
+import com.example.onlinebookshop.exceptions.EntityNotFoundException;
+import com.example.onlinebookshop.exceptions.TooManyObjectsException;
+import com.example.onlinebookshop.mapper.ShoppingCartMapper;
+import com.example.onlinebookshop.repositories.book.BookRepository;
+import com.example.onlinebookshop.repositories.cartitem.CartItemRepository;
+import com.example.onlinebookshop.repositories.shoppingcart.ShoppingCartRepository;
+import com.example.onlinebookshop.services.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCartByUserEmail(Long userId) {
+        return shoppingCartMapper.toShoppingCartDto(shoppingCartRepository.findByUserId(userId));
+    }
+
+    @Override
```

Comment:
Don't forget about `@Transactional`. There are more than one request to DB. Check other places